src/transformers/training_args.py

```diff
@@ -444,6 +445,16 @@ class TrainingArguments:
     _n_gpu: int = field(init=False, repr=False, default=-1)
 
     def __post_init__(self):
+        if self.output_dir is None and os.getenv("SM_OUTPUT_DATA_DIR") is None:
+            raise ValueError(
+                "`output_dir` is only optional if it can get inferred from the environment. Please set a value for "
+                "`output_dir`."
+            )
+        elif os.getenv("SM_OUTPUT_DATA_DIR") is not None:
```

Comment:
Shouldn't this check is `self.output_dir is not None` before logging a warning that it's being overwritten?

Reply:
Ah yes, good point!